pkg/test/integration/common/framework.go

```diff
@@ -0,0 +1,815 @@
+	for i := range containers {
+		// splitedString holds image name in splitedString[0]
+		var splitedString []string
+		if strings.Contains(containers[i].Image, "@") {
+			splitedString = strings.Split(containers[i].Image, "@")
+		} else {
+			splitedString = strings.Split(containers[i].Image, ":")
+		}
+
+		if providerSpecificRegexp.Match([]byte(containers[i].Image)) {
+			// set container image to mcContainerImageTag as the name of the container contains provider
+			if len(mcContainerImageTag) != 0 {
+				containers[i].Image = splitedString[0] + ":" + mcContainerImageTag
+			}
+		} else {
+			// set container image to mcmContainerImageTag as the name of container contains provider
+			if len(mcmContainerImageTag) != 0 {
+				containers[i].Image = splitedString[0] + ":" + mcmContainerImageTag
+			}
+
+			// set machine-safety-overshooting-period to 300ms for freeze check to succeed
+			var isOptionAvailable bool
+			for option := range containers[i].Command {
+				if strings.Contains(containers[i].Command[option], "machine-safety-overshooting-period=") {
+					isOptionAvailable = true
+					containers[i].Command[option] = "--machine-safety-overshooting-period=300ms"
+				}
+			}
+			if !isOptionAvailable {
+				containers[i].Command = append(containers[i].Command, "--machine-safety-overshooting-period=300ms")
+			}
+		}
+	}
+
+	// apply updated containers spec to mcmDeploymentObj in kubernetes cluster
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		// Retrieve the latest version of Deployment before attempting to update
+		// RetryOnConflict uses exponential backoff to avoid exhausting the apiserver
+		mcmDeployment, getErr := c.ControlCluster.Clientset.AppsV1().Deployments(controlClusterNamespace).Get("machine-controller-manager", metav1.GetOptions{})
+		if getErr != nil {
+			log.Printf("failed to get latest version of Deployment: %v", getErr)
+			return getErr
+		}
+		mcmDeployment.Spec.Template.Spec.Containers = containers
+		_, updateErr := c.ControlCluster.Clientset.AppsV1().Deployments(controlClusterNamespace).Update(mcmDeployment)
+		return updateErr
+	})
+
+	ginkgo.By("Checking controllers are ready in kubernetes cluster")
+	gomega.Eventually(func() int {
+		deployment, err := c.ControlCluster.Clientset.AppsV1().Deployments(controlClusterNamespace).Get("machine-controller-manager", metav1.GetOptions{})
+		if err != nil {
+			log.Println("Failed to get deployment object")
+		}
+		return int(deployment.Status.ReadyReplicas)
+	}, 60, 5).Should(gomega.BeNumerically("==", 1))
+
+	return retryErr
+}
+
+func (c *IntegrationTestFramework) scaleMcmDeployment(replicas int32) error {
+	/*
+		 - if any of mcmContainerImage  or mcContainerImageTag flag is non-empty then,
+			 update machinecontrollermanager deployment in the control-cluster with specified image
+		 -
+	*/
+	result, getErr := c.ControlCluster.Clientset.AppsV1().Deployments(controlClusterNamespace).Get("machine-controller-manager", metav1.GetOptions{})
```

Comment:
Can we also change the deployment name to an ENV variable like `MCM_DEPLOYMENT_NAME`?

Reply:
This has been done as suggested. Thanks a lot for the input